lib/foreman/telemetry_sinks/prometheus_sink.rb

```diff
@@ -1,7 +1,14 @@
 module Foreman::TelemetrySinks
   class PrometheusSink
+    PROMETHEUS_STORE_DIR = File.join(Rails.root, 'tmp', 'prometheus')
+
     def initialize(opts = {})
       require 'prometheus/client'
+      require 'prometheus/client/data_stores/direct_file_store'
+      # Set multiprocess-friendly data store
+      FileUtils.mkdir_p(PROMETHEUS_STORE_DIR)
```

Comment:
Instead of a global, should this be `store_dir = opts[:store_dir] || File.join(Rails.root, 'tmp', 'prometheus')`?

Reply:
I don't like introducing new option, then setting, then installer. Let's be opinionated here - these files will rarely be larger than few MBs so Rails cache is an ideal place. You can lost them and it gets regenerated on restart, no big deal too.

Reply:
So this is a parameter to the method, which also allows unit testing. I wasn't calling for it to be configurable at the setting/installer level.

Reply:
I agree the constant is not necessary as it doesn't need to be this exact dir, it's only a sane default, so why not to have the method more robust. I'm ok with this as well though.